CodeEdit/Features/Documents/Controllers/CodeEditDocumentController.swift

```diff
@@ -8,6 +8,42 @@
 import Cocoa
 
 final class CodeEditDocumentController: NSDocumentController {
+    lazy var fileManager: FileManager = {
+        FileManager.default
+    }()
+
+    override func newDocument(_ sender: Any?) {
+        guard let newDocumentUrl = self.newDocumentUrl else { return }
+
+        let createdFile = self.fileManager.createFile(
+            atPath: newDocumentUrl.path,
+            contents: nil,
+            attributes: [FileAttributeKey.creationDate: Date()]
+        )
+        guard createdFile else {
+            print("Failed to create new document")
+            return
+        }
+
+        self.openDocument(withContentsOf: newDocumentUrl, display: true) { _, _, _ in }
+    }
+
+    private var newDocumentUrl: URL? {
+        guard let documentsUrl = self.fileManager.urls(for: .documentDirectory, in: .userDomainMask).first else {
+            print("Unable to get user's Documents directory")
+            return nil
+        }
+
+        var fileNumber = 0
+        var fileUrl = documentsUrl.appendingPathComponent("Untitled")
+        while self.fileManager.fileExists(atPath: fileUrl.path) {
+            fileNumber += 1
+            fileUrl = fileUrl.deletingLastPathComponent().appendingPathComponent("Untitled\(fileNumber)")
+        }
+
```

Comment:
Could we use a [`NSSavePanel`](https://developer.apple.com/documentation/appkit/nssavepanel) here instead? It would look something like:
```
let panel = NSSavePanel()
// Maybe add some customization to the panel here...
guard panel.runModal() == .OK else {
    return nil
}

return panel.url
```
It would allow users to give a file name, extension and directory to save in a quick panel. You mentioned the alternative of letting users select a file type, but an `NSSavePanel` is what's default on macOS.

Reply:
Is this when the user creates a new file (cmd+N)? I like VS Code’s approach. Create an untitled file that isn’t saved in the project until the user saves. The question is then where does this temporary file get saved because it should persist when project is closed and reopened. There can also be multiple untitled files open at once. I’d imagine they’d all be labeled “Untitled” but perhaps stored uniquely. Thoughts?

Reply:
Our cmd+N behaves very different when compared to VSCode right now. Right it just opens a new window with a new file. On VSCode cmd+N creates a new file in the project window and cmd+shift+N creates a new project window

Reply:
@matthijseikelenboom are you saying you want it to behave differently than VS Code?

Reply:
No I'm saying that it currently does not behave like VSCode. And that I don't know if it has been discussed that way, or that the current implementation is just a placeholder. If it hasn't been discussed, perhaps we should do that. But in a different place then on this PR.

Reply:
Do we want to make this change as part of this PR then?